### ISSUE TITLE
audit-engine: validateByRuleKeys() walks the MERGED config, so RuleOfCode's own default ignore keys are reported to the user as their unknown laws

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#140]
**Version:** 7.23.0
**Law:** not a law — `AuditEngine.validateByRuleKeys()` in `dist/core/auditing/audit-engine.js`, reading the defaults in `dist/config/types.js`

Kind: it reports a problem the user cannot act on — the keys it names are RuleOfCode's own, not the project's.

## What happens

Five `unknown law` warnings on every run, naming keys that do not exist anywhere in the project:

```
⚠️  ignores.byRule: unknown law 'angular-standalone-components' — these patterns filter nothing.
⚠️  ignores.byRule: unknown law 'angular-onpush-strategy' — these patterns filter nothing.
⚠️  ignores.byRule: unknown law 'angular-trackby-functions' — these patterns filter nothing.
⚠️  ignores.byRule: unknown law 'angular-async-pipe' — these patterns filter nothing.
⚠️  ignores.byRule: unknown law 'ngrx_d494c2aa' — these patterns filter nothing.
```

They are not in the config:

```
$ grep -c "angular-async-pipe" ruleofcode.config.json
0
$ node -e "const c=require('./ruleofcode.config.json'); console.log(Object.keys(c.ignores.byRule))"
[ '$comment', 'code-duplication-control', 'centralized-logging',
  'environment-security-enforcement', 'environment-parity-standards',
  'zero-tolerance-doctrine', 'absolute-zero-tolerance-doctrine',
  'test-coverage-constitutional-standard', 'error-handling-completeness', ... ]
```

## Reproduction

Run the audit with `ignores.byRule` and `includes.byRule` emptied completely. The five warnings survive:

```
$ node -e "const c=require('./ruleofcode.config.json'); c.ignores.byRule={}; c.includes.byRule={};
  require('fs').writeFileSync('empty.json', JSON.stringify(c,null,2))"

$ npx ruleofcode audit --mode=full --config empty.json | grep -c "unknown law"
5
```

Nothing the user writes can silence them, because they are not the user's.

## Where they come from

`dist/config/types.js` ships a default `ignores.byRule` that is merged into the user's config:

```js
'angular-async-pipe': [
    `${constants_1.CONSTITUTIONAL_PATHS.RULEOFCODE_PACKAGE}/**`,
    `${constants_1.CONSTITUTIONAL_PATHS.CONSTITUTIONAL_COMPLIANCE_PATTERN}/**`,
],
angular_95b77e5f: [ ... ],
ngrx_e156533b: [ ... ],
```

`templates/angular.config.js` carries the same family:

```js
'angular-standalone-components': true,
'angular-onpush-strategy': true,
'angular-trackby-functions': true,
'angular-async-pipe': true,
```

Then `validateByRuleKeys()` walks the merged map and warns for every key that does not resolve — including the ones it just merged in itself:

```js
for (const key of Object.keys(section)) {
    if (/^law-\d+$/.test(key)) continue;
    if (known.has(key)) continue;
    console.warn(`⚠️  ${label}: unknown law '${key}' — these patterns filter nothing. ...`);
}
```

None of the five resolves against the registry:

```js
const {ModularLawsRegistry: R} = require('ruleofcode/dist/registry/modular-laws-registry.js');
const keys = new Set();
for (const l of R.getAll()) for (const k of R.lawIdentityKeys(l)) keys.add(k);
// 169 laws, 717 identity keys
// 'angular-async-pipe' ∈ keys → false   (same for the other four)
```

## What it cost us

The warning is genuinely useful — in the same output it correctly caught six stale names in our own config, and deleting those was a real improvement. But a team ticket was opened to "fix all eleven", and five of the eleven have no fix available in our repo. Someone had to grep two config files, then the package, then run the audit against an emptied config, to establish that five of them are yours and not ours. The warning text — "these patterns filter nothing. Check the name with 'roc laws --list'" — points the reader at their own config, which is the one place the answer is not.

This is the same class as #130: false positives sitting beside true ones teach people to skim the block, and skimming is how the true ones survive.

## Suggested fix

Either of these; the first is smaller:

1. Validate only keys the **user** supplied — check the parsed config before the defaults are merged, so the engine never reports its own entries.
2. Correct the default names so they resolve. `angular-onpush-strategy` and `angular-standalone-components` look like they mean `onpush-change-detection-strategy` and `standalone-components-architecture`, which do exist in the registry; `angular-async-pipe`, `angular-trackby-functions` and `ngrx_d494c2aa` have no obvious counterpart, so this path needs a decision per key rather than a rename.

If the defaults are intentionally forward-looking placeholders for laws not yet written, then (1) is the right answer regardless — a placeholder should not be announced to the user as their mistake.

## What I did NOT verify

- Whether `templates/angular.config.js` is actually loaded at audit time, or only by `roc init`. I established that `dist/config/types.js` supplies the merged defaults; the template file merely carries the same names.
- Whether the same shape exists for other stacks (python, node). I only read the Angular defaults, because that is the project type here.
- The exact provenance of `ngrx_d494c2aa` and `angular_95b77e5f` — hash-suffixed ids suggest generated names, and I did not trace how they are produced.
